### ISSUE TITLE
Using the limit query argument with the tests api endpoint causes a 504

### ISSUE THREAD

**Issue by jscook2345** [Linaro/squad#698]
https://qa-reports.linaro.org/api/tests/?test_run=2553969&&limit=1

Results in: 504 Gateway Time-out

@danrue

**Comment by mwasilew** [comment]
@jscook2345 tests endpoint doesn't use offset/limit pagination so limit has no meaning.

**Comment by jscook2345** [comment]
@mwasilew 

It appears to work when `limit` is set to another value.

example:

https://qa-reports.linaro.org/api/tests/?test_run=2553969&&limit=5

and

https://qa-reports.linaro.org/api/tests/?limit=5

both work as expected.

**Comment by mwasilew** [comment]
That's weird. Let me read the docs so I can discover how it works. In my understanding limit should not have any meaning here.

**Comment by mwasilew** [comment]
OK, found it. @chaws made a custom pagination class: https://github.com/Linaro/squad/blob/master/squad/api/utils.py#L12

**Comment by mwasilew** [comment]
Right, the problem might be with the database size. Same query works well on staging:
https://staging-qa-reports.linaro.org/api/tests/?limit=1

**Comment by chaws** [comment]
@mwasilew I don't think it's database size related because it brings results back in less than a second for limit > 2:

Works alright:
https://qa-reports.linaro.org/api/tests/?test_run=2553969&limit=3

Times out:
https://qa-reports.linaro.org/api/tests/?test_run=2553969&limit=2

Just pick 5 at a time @jscook2345 :p 

I'll take a look into that tomorrow

**Comment by jscook2345** [comment]
Not sure if it is related or not, but here is another query against the `tests` endpoint that results in a 504.

https://qa-reports.linaro.org/api/tests/?ordering=-id&&suite=144

If you use just one of the query parameters (either one) it works fine.
